Feature/can

mvillalbap opened the pull request.

mvillalbap committed:
Provisional development

mvillalbap committed:
Receiving CAN messages

Next step to recode the structure and quit unused variables

mvillalbap committed:
Cleaner version of CAN support

mvillalbap committed:
Changes in MCP2515 driver and SPI-DMA hal

There are also minor changes in FreeRTOS Trace names of the tasks. They are now defines due to errors thrown and efficiency in the preprocessor

mvillalbap committed:
Solved wake up from MCP2515 problem

Nowadays, the system is not enough intelligent to know a wake up from the CAN line, only it can send CAN frames if user sets normal operation mode

mvillalbap committed:
Merge branch 'develop' of boxfish-studio/totem into feature/can

agus-xyz committed:
fixed classpath of gui app

agus-xyz committed:
minor format and comment changes

agus-xyz committed:
added totem can demo app

agus-xyz reviewed (changes requested):
for some reason some not all can messages are received, and those which are received come with delay. I added a can demo app to help debugging can message reception

mvillalbap committed:
Fixed receive demo CAN application

The host can down to 5 ms each message sent. With 3 it starts to lose messages and with 1 it freezes

agus-xyz committed:
minor changes

agus-xyz committed:
added dummy time sending over can for demo

agus-xyz commented:
please lets resolve the conflicts with the develop

mvillalbap committed:
Fixed error sending over CAN

Make a SPI_DMA_Handler in order to reuse the structure

mvillalbap committed:
Factorized the SPI using the DMA

Next, comment the MCP2515 driver and the HAL SPI with DMA

mvillalbap committed:
Added comments to MCP2515 driver and SPI with DMA hal

mvillalbap committed:
Merge branch 'develop' of github.com:boxfish-studio/totem into feature/can

mvillalbap committed:
Remove Source folder from previous merge

agus-xyz committed:
minor comments and format changes

agus-xyz committed:
renamed and reallocated host apps

agus-xyz reviewed (approved):
this looks amazing
